### ISSUE TITLE
Consolidate --show-* options

### ISSUE THREAD

**Issue by dpb587** [dpb587/gget#13]
The `--show-ref` and `--show-resources` cannot easily be used together (since they have different formats), and they cannot be used in the same command as a download. They're primarily intended for debugging and automation. Find a way to support a single command to download files while still having a method to capture metadata useful for external tools.

Maybe introduce a `--print-info`-like option with a machine-friendly output format like below. It would be helpful to be support sending this to a non-STDOUT/STDERR descriptor to continue to maintain a nice user output:

```
origin	resolved	github.com/dpb587/gget@v0.4.0
origin	service	github
origin	server	github.com
origin	owner	dpb587
origin	repository	gget
origin	ref	v0.4.0
metadata	tag	v0.4.0
metadata	commit	d75361c1316c6004cde13c04460c57552a839d57
resource	gget-0.4.0-darwin-amd64
resource	gget-0.4.0-linux-amd64
resource	gget-0.4.0-windows-amd64.exe
```

Probably requires introducing a dedicated `--no-download`-like option to allow external tools to continue gathering info before an actual download.

**Comment by dpb587** [comment]
I added `--no-download`, and went with `--dump-info=` since it kind of matches `curl`'s `--dump-header`. The `--show-ref` and `--show-resources` flags are now hidden from the help text and I intend to remove them before a 1.x version (after which commands will error with the flags being unrecognized).

I'm still thinking if I want to change the `resource` prefix/format of files in case other info might need to be shown per file at some point (e.g. size).

@whalehub, I'm typically not too concerned about backwards compatibility before a 1.0, but figured I'd call it out since I think you're relying on `--show-ref`. I'll keep around those old options for a couple minor versions. Let me know if you have feedback on the new options or output. The below are some examples which may help you switch once they're available (note: it'll probably be a week before I get back to this and potentially cut a new release).

```
# detect if --dump-info is supported
$ gget --version='>=0.5' --quiet 2>/dev/null

# getting the resolved version
$ go run . github.com/dpb587/gget --no-download --dump-info=- \
  | awk '{ if ( $1 == "origin" && $2 == "ref" ) { print $3 } }'
v0.4.0

# listing resource names
$ go run . github.com/dpb587/gget --no-download --dump-info=- \
  | awk '{ if ( $1 == "resource" ) { print $2 } }'
gget-0.4.0-darwin-amd64
gget-0.4.0-linux-amd64
gget-0.4.0-windows-amd64.exe
```

**Comment by ghost** [comment]
@dpb587 I don't have any objections to the new format since all it takes is a small change in my script from

```bash
gget --show-ref "${!REPO}" 2>/dev/null | sed '2d;s|tag[[:blank:]]||g;s|v||g')"

```

to

```bash
gget --no-download --dump-info=- "${!REPO}" 2>/dev/null | grep metadata | sed '2d;s|.*tag[[:blank:]]||g;s|v||g')"
```

before everything is back to normal again:

```
whalehub@pdh:~# ginstall.sh -c gget
The latest version of gget is v0.4.0.
```

I'll wait for you to cut a new release before I publish the next version of `ginstall.sh` since these changes will require bumping the minimum required version of `gget` for my script.

**Comment by ghost** [comment]
@dpb587 Just tried out gget v0.5.0 and it works quite nicely. Thanks for your hard work! 🎉

```bash
VERSION_CHECK="$(gget --no-download --export plain "${!REPO}" 2>/dev/null | grep metadata | sed '1,2d;s|.*tag[[:blank:]]||g;s|v||g')"
if [ -z "${VERSION_CHECK}" ]; then
  APP_VERSION="$(gget --ref-stability=pre-release --no-download --export plain "${!REPO}" 2>/dev/null | grep metadata | sed '1,2d;s|.*tag[[:blank:]]||g;s|v||g')"
else
  APP_VERSION="${VERSION_CHECK}"
fi
```

```sh
whalehub@pdh:~# ginstall.sh -c gget
The latest version of gget is v0.5.0.
whalehub@pdh:~# sudo ginstall.sh gget latest
Found 1 file (11.1M) from github.com/dpb587/gget@v0.5.0
√ gget-0.5.0-linux-amd64 done (sha256 OK; executable)
The latest version of gget (v0.5.0) was successfully installed to /usr/local/bin.
```

**Comment by dpb587** [comment]
One other thing I changed since my last comment was the export, as you found, but it also now supports jsonpath-ish. It should be a bit easier to extract the ref (and use fewer API requests than `plain` in a future optimization). Something like the following should be equivalent:

```
for stability in stable pre-release ; do
  APP_VERSION="$(gget --ref-stability="${stability}" --export jsonpath='{.origin.ref}' --no-download "${!REPO}" 2>/dev/null | sed 's|^v||')"
  [ -z "${APP_VERSION}" ] || break
done
```

I'd encourage using the origin ref instead of metadata tag since that's what gget is technically referring to and prefers. Also I've learned I need to namespace those metadata keys and include some additional metadata in an upcoming version; will make an issue.
